mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
             kind='average')  # nave and kind are technically incorrect
         cumul += tmax_ - tmin_
     return evokeds
+
+
+class EncodingModel(object):
+    def __init__(self, delays=None, est=None, preproc_y=True):
+        """Fit a STRF model.
+
+        Fit a receptive field using time lags and a custom estimator or
+        pipeline. This implementation uses Ridge regression and scikit-learn.
+        It creates time lags for the input matrix, then does cross validation
+        to fit a STRF model.
+
+        Parameters
+        ----------
+        delays : array, shape (n_delays,)
+            The delays to include when creating time lags. The input array X
+            will end up having shape (n_feats * n_delays, n_times)
```

Comment:
Could you give an example of a transformer that does this? IIUC, you'd multiply delays by sfreq regardless. That way if:

* sfreq is 1, nothing changes so we assume that delays is in indices.
* sfreq is != 1, then we are assuming that the input was actually in seconds and we are multiplying by a sampling frequency.

Does that sound correct? In that case, maybe it makes sense to throw a warning if `sfreq != 1` and `delays.dtype == int`.